src/drive/components/FileListPlaceholder.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+import styles from '../styles/table'
+
+// using a seeded PRNG to prevent re-renders from changing the results
+const seededRandom = seed => {
+  const x = Math.sin(seed) * 10000
+  return x - Math.floor(x)
+}
+const seededRandomBetween = (min, max, seed) =>
+  min + seededRandom(seed) * (max - min)
+
+const PlaceholderBlock = ({ width }) => (
+  <div className={styles['fil-content-file-placeholder']} style={{ width }} />
+)
+
+PlaceholderBlock.propTypes = {
+  width: PropTypes.string
+}
+
+PlaceholderBlock.defaultProps = {
+  width: '100%'
+}
+
+const FilePlaceholder = ({ index }) => (
+  <div className={styles['fil-content-row']}>
+    <div
+      className={classNames(
+        styles['fil-content-cell'],
+        styles['fil-content-file-select']
+      )}
+    />
+    <div
+      className={classNames(
+        styles['fil-content-cell'],
+        styles['fil-content-file']
+      )}
+    >
+      <PlaceholderBlock width={`${seededRandomBetween(3, 12, index)}rem`} />
+    </div>
+    <div
+      className={classNames(
+        styles['fil-content-cell'],
+        styles['fil-content-date']
+      )}
+    >
+      <PlaceholderBlock width="5rem" />
+    </div>
+    <div
+      className={classNames(
+        styles['fil-content-cell'],
+        styles['fil-content-size']
+      )}
+    >
+      <PlaceholderBlock width={`${seededRandomBetween(3.75, 5, index)}rem`} />
+    </div>
+    <div
+      className={classNames(
+        styles['fil-content-cell'],
+        styles['fil-content-status']
+      )}
+    >
+      <PlaceholderBlock width="1.25rem" />
+    </div>
+  </div>
+)
+
+FilePlaceholder.propTypes = {
+  index: PropTypes.number
+}
+
+FilePlaceholder.defaultProps = {
+  index: 1
+}
+
+const FileListPlaceholder = ({ rows }) => (
+  <div>
+    {Array.apply(null, Array(rows)).map((value, index) => (
```

Comment:
I had hard time to understand this code and I still can't understand why we have to use `Array.apply(null, new Array(5))`.

But I found a clearer write:

```js
[...new Array(rows)].map(createFilePlaceholder)
```

Reply:
Oh that *is* better, nice. I tried some variations like that but couldn't find one that worked. Using that instead 👍